### PR TITLE
First request comments

### PR DESCRIPTION
**Pull request by iznv**

### CODE REVIEW

File: VK Feed/Cells/FeedCell/FeedCell.swift [hunk 1]
```diff
@@ -27,10 +28,8 @@ class FeedCell: UITableViewCell, ConfigurableCell {
...
         
-        if let imageUrl = post.imageUrl {
-            let photo = (post.attachments?.first?.photo)!
-            imageImageView.stretchImageToScreenWidthKeepingAspectRatio(imageSize: (photo.height!, photo.width!))
-            
+        if let imageUrl = post.imageUrl, let photo = post.attachments.first?.photo {
+            imageImageView.stretchImageToScreenWidthKeepingAspectRatio(imageSize: (photo.height, photo.width))
             imageImageView.sd_setImage(with: URL(string: imageUrl))
         }
         else {
```
Comment (NikAshanin) [inline VK Feed/Cells/FeedCell/FeedCell.swift]: if {
} else {
}

File: VK Feed/Controllers/Feed/View/FeedViewController.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    
+    // MARK: viewDidLoad
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureRefreshControl()
+        
+        if !Reachability.isInternetAvailable {
+            addPostsToTableView(VKWallPost.getFromRealm(), allowLoadMore: false)
+        }
+        else {
```
Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: same about else

File: VK Feed/Controllers/Feed/View/FeedViewController.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        tableView.addSubview(refreshControl)
+    }
+    
+    func refresh(_ refreshControl: UIRefreshControl) {
+        if Reachability.isInternetAvailable {
+            tableDirector.clear()
+            viewModel.clearWallPosts()
+            getWallPosts()
+        }
+        else {
```
Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: same

File: VK Feed/Controllers/Feed/View/FeedViewController.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            refreshControl.endRefreshing()
+        }
+    }
+    
+    func getWallPosts() {
+        viewModel.getWallPosts() {
+            wallPosts in
+            self.addPostsToTableView(wallPosts)
```
Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: redundant self

File: VK Feed/Controllers/Feed/View/FeedViewController.swift [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    
+    // MARK: Methods
+    func configureRefreshControl() {
+        refreshControl.addTarget(self, action: #selector(refresh), for: .valueChanged)
+        tableView.addSubview(refreshControl)
+    }
+    
+    func refresh(_ refreshControl: UIRefreshControl) {
+        if Reachability.isInternetAvailable {
```
Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: should be moved to viewmodel

File: VK Feed/Controllers/Feed/View/FeedViewController.swift [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            viewModel.clearWallPosts()
+            getWallPosts()
+        }
+        else {
+            refreshControl.endRefreshing()
+        }
+    }
+    
+    func getWallPosts() {
+        viewModel.getWallPosts() {
```
Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: should be moved to viewmodel

File: VK Feed/Models/API/HTMLError.swift [hunk 7]
```diff
@@ -0,0 +1,11 @@
+//
+//  HTMLError.swift
+//  VK Feed
+//
+//  Created by Ivan Zinovyev on 12/7/16.
+//  Copyright © 2016 Ivan Zinovyev. All rights reserved.
+//
+
+enum HTMLError: Int {
+    case AmetCookieValidation = 401
```
Comment (NikAshanin) [inline VK Feed/Models/API/HTMLError.swift]: AmetCookie?) mm? it's just for Drive2

File: VK Feed/Controllers/Feed/View/FeedViewController.swift (hunk 4 above)
Comment (iznv) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: actually not. this is a closure

Comment (NikAshanin) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: so weak self should be here

Comment (iznv) [inline VK Feed/Controllers/Feed/View/FeedViewController.swift]: ok

**Pull request merged**